torch/backends/cuda/__init__.py

```diff
@@ -164,6 +165,19 @@ def preferred_linalg_library(backend: Union[None, str, torch._C._LinalgBackend]
     return torch._C._get_linalg_preferred_backend()
 
 
+class SDPBackend(IntEnum):
+    r"""Enum class for the scaled dot product attention backends.
+
+    .. warning:: This flag is experimental and subject to change.'
+
+    This class needs to stay inline with the enum defined in:
+    pytorch/aten/src/ATen/native/transformers/sdp_utils_cpp.h
+    """
+    MATH = 0
+    FLASH_ATTENTION = 1
+    EFFICIENT_ATTENTION = 2
+    ERROR = 3
```

Comment:
Since this integer is signed, an error could also be "-1".

Reply:
Then we don't have to do the following in the future

```
[...]
    EFFICIENT_ATTENTION = 2
    ERROR = 3
    NEW_ATTENTION_2 = 4
```

Reply:
great idea, just changed